modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/auto_fields.js

```diff
@@ -167,7 +167,7 @@ AUI.add(
 
 						trigger.placeAfter(list);
 
-						list.append(palette);
+						list.append(palette.cloneNode(true));
```

Comment:
Seeing as the palette has an ID, won't this create a duplicate element in the DOM (ie. with a clashing ID)? I don't know what this file does, so I'm lacking context here. Are you able to provide some explanation of how/why this fixes the LPS?

Reply:
Hi @wincent , 
Multiple elements can be added using Auto Field. We use it for cloning elements, so the elements will be duplicated. The problem would be if the IDs were duplicated, too, but it does not the case. I checked it: when I add new fields, each will get a unique ID in the DOM. The fix is about not only copying the palette, but deep copy all of the content inside of that. Therefore, we do not get an empty dropdown element only, but also the containing list items. Let me know if there is any comment on this or any concern. Thank you!

**Empty dropdown:** 
![ExampleEmpty](https://user-images.githubusercontent.com/69793575/115286637-91784a00-a14f-11eb-822c-1cd661e045d6.JPG)

**After the fix:**
![Example](https://user-images.githubusercontent.com/69793575/115286895-eb790f80-a14f-11eb-80c1-e1b0ce79e4e6.JPG)

Reply:
Just to add to @orsolyaDekany 's comment the reason why the id is unique even after deep copy it is because we manipulated later. You can see we invoke [_createCloneFromMarkup](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/auto_fields.js#L176-L182) function, which updates the name using an index, and later uses that name as an id. So the `id` is going to include an index which will make it unique.

About why we need this change, once we open the menu for the first time the `ul` node containing the list is moved from the row to the top of the body. So when we copy the row, we need to add that [extra step](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/auto_fields.js#L157-L171) recovering the list from the top of the body (`paletteIsClone` variable is telling us the menu we are copying had been opened). 

If we just do `list.append(palette)` we'll later lose the reference for the first menu when trying to retrieve it again from [_getMenu](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/menu.js#L197) and we'll show an empty list. Performing a deep copy we'll keep both references.

Reply:
Thank you both for the explanation.

Test failure seems spurious, so going to try forwarding.